### ISSUE TITLE
FMPro # entry field & view in Configurator & generator

### ISSUE THREAD

**Issue by GoogleCodeExporter** [willdwyer/bcbsmaissuestracker#218]
```
Need to track our FMPro # in Highroads given workflow was de-scoped. See 
attached screen shots for sample layout and preferences. Can it be added as an 
attribute so it can be used in search?

Configurator Screen
- Entry field in plan details above Preferred Acct Name
- Entry Field label: FMPro #
- 6 digit # will be typed into field 
- If more than 6 digits entered, need error to indicated incorrect # to user
- Display to right of HR # & Acct & Name in header once entered & saved

Generation Screen
- Display to right of HR # & Acct & Name in header

Can FMPro# be added as an attribute for custom only
- Content Library would display to the left of the document # column
- Sortable
- Searchable


```

Original issue reported on code.google.com by `decoste....@gmail.com` on 30 Jun 2015 at 5:39

Attachments:
- [FMPro_No_enhancement.pdf](https://storage.googleapis.com/google-code-attachments/bcbsmaissuestracker/issue-218/comment-0/FMPro_No_enhancement.pdf)

**Comment by g4gaurang** [comment]
#20 michelle.at.highroads
yes, if we reuse the existing Stock Number field, since this is not used for Custom plans, then that would cut down on the LOE. New effort would include adding this field into the configurator as an editable field and adding validation rules
Issues we would run into: the error messages for this field having the wrong plan design field name (stock number vs FMPro#).

**Comment by g4gaurang** [comment]
Based on 9/23 meeting, HR to assess the impact of HR job number on print workflow and distribution and come back to BCBSMA. Michelle, let us know if we can document any requirement updates based on our 9/23 meeting once you come back with assessment.

**Comment by michparrish** [comment]
We revamped the original requirements document and included the additional items discussed yesterday. I sent this document to Jenn D this morning, along with my Distribution and WF test results, for your team's review and approval.
I can attach that doc here once I learn how. I don't see an 'add attachment' button.

**Comment by g4gaurang** [comment]
No need to attach the document here if you already sent it to Jenn. We will review Monday morning and get back to you.

**Comment by markderg** [comment]
Waiting on final approval of enhancement from BCBSMA.  Current estimate is 25 hours.  Jen DeCoste has e-mail overview from M. Parrish. Once approval is given HR will begin work.

**Comment by jddecoste** [comment]
BCBSMA reviewed doc and provided updates, including allowing add of "BS" to theFMPro name, ensuring required field, ensuring FMPro number and convention described in doc will show as the file doc name and plan design name. Added notation that the HR # should also appear on the last page of the output SBC aligned to the left of the reg wording.  Waiting on updated requirements doc to be sent and then approval can be provided

**Comment by jddecoste** [comment]
Updated requirements doc rec'd 7/27 and approved by BCBSMA 7/28.
Please confirm the LOE estimate remains at 25-30 hours

thank you

**Comment by michparrish** [comment]
Confirmed - LOE is estimated at 25-30hrs of work inclusive of design, configuration, testing and release to Prod.

**Comment by michparrish** [comment]
FMPro# and all requirements have been updated and tested to success in pclient5.
this item is now ready for testing by BCBSMA team. @g4gaurang @willdwyer

**Comment by g4gaurang** [comment]
**Actual Result:** See screenshots below. When the FMPro# is updated on a plan design the second time ( assume user entered incorrect FMPRO# first time), 
1. The newest version of the plan design naming convention correctly updates to the newer FMPRO# but the previous versions naming conventions also now show up  with the newer FMPRO#
2. If you probe further, you will notice that the previous plan design version has the newer FMPRO# in the naming convention but inside the plan design, the FMPRO# is the old one.
3. The document generated still carries the first FMPRO# even when the plan is regenerated after changing to the new FMPRO#.

**Expected Result:** 
1.  The FMPRO# on the plan design naming convention should match the FMPRO# captured inside that plan design.
2. The document naming convention should  update to the newer FMPRO#.

![image](https://cloud.githubusercontent.com/assets/10272072/9122157/0eed2cc8-3c55-11e5-8af1-5e0488cdce27.png)

![image](https://cloud.githubusercontent.com/assets/10272072/9122318/400d6f10-3c56-11e5-8436-6e029685a6b8.png)

![image](https://cloud.githubusercontent.com/assets/10272072/9122543/bdb7d45e-3c57-11e5-8d9e-baa3dfb24a1c.png)

**Comment by michparrish** [comment]
@g4gaurang 
1. correct - this is similar to updating the name of a table in Access or renaming a file in Word, Excel, etc. We do not keep historical names of plan designs, only history of the data within. So if you change the FMPro# of a plan design, then the  naming convention will change as well, for all versions b/c that part isn't versioned.
2. correct - this is because of versioning the data within the plan design and you can see happens in all data update cases. If you had $35 copayment in the June 18, 2015 version of a doc and updated to $45 on July 2nd, then you'd still have that previous version available to you showing the $35 and a new version showing $45. 
3. correct - this should be the only 'manual' point of update for the team - the documents generated from the plan design, though the plan design has been updated, will continue to carry the old value UNLESS the original documents are manually renamed with the new FMPro#. So an additional step, should there be a situation where the user has already generated documents from a plan design that had an incorrect original FMPro#, should be to find the original documents and use Tools > Rename to edit the name of the documents. Then any further documents generated from the now-updated plan design will carry the correct name (this was tested prior to posting this to be sure). 

Please let me know if there are any concerns or questions on the above, or any additional items found in testing.

**Comment by michparrish** [comment]
we can take back the idea of breaking the link between the newly named plan design and the old set of incorrectly named SBCs, to have the newly named plan design generate NEW documents with the correct name

**Comment by willdwyer** [comment]
We encountered a new issue with the FMPro#.  'In-Network' appears next to FMPro# in Advantage Blue Plan.

This appears by recreating the following steps:
1. On Dashboard, click Manage Plan
2. On Search (For All Plans) screen, type in following: 
   a. Effective Date: 2016 
   b. Product Group Type: Advantage Blue 
   c. Plan Type: Standard
3. Click Show Me Plans button
4. On search results screen, hover cursor over plan
5. Click ‘Create a New Plan’ button
6. Click Misc Fields tab
7. See screenshot

Please fix globally.

![screen shot 2015-08-13 at 4 22 29 pm](https://cloud.githubusercontent.com/assets/13453008/9261289/dc8de9da-41da-11e5-9375-91496c82c817.png)

**Comment by michparrish** [comment]
this has been fixed - this only impacted the EPO plan. Please continue to retest. Thank you.

**Comment by michparrish** [comment]
@jddecoste Regarding the decoupling of a renamed plan design with the name of the documents that are generated from it, I have to submit an enhancement and it will go through our feasibility/estimation process. Please confirm that you'd like me to submit this.
Workaround is for the user to delete or rename the documents already created from the incorrectly-named plan design and regenerate with the new name.

**Comment by g4gaurang** [comment]
@michparrish , This is an HR system limitation which prevents the user from fully realizing the intended benefits of the  FMPRO feature requirement. The proposed manual workaround kills any efficiency benefit of the process and just transfers work effort from the system to the user. We would like this bug to be resolved and would like to get a  feasibility and LOE estimate. 

CC @jddecoste

**Comment by markderg** [comment]
Gaurang/Jenn D.

I ask we have a 1 hour meeting to review this original enhancement request, what was met from a requirement stand point and determine any next steps.  

Please advise us on dates and times that work for you and I can send out a meeting invite.  

Thx, Mark

**Comment by michparrish** [comment]
@g4gaurang @willdwyer @jddecoste 
I confirmed with Prod today that this will be an enhancement request, the system is working as designed. I want to note that Prod will not be able to schedule the enhancement to a sprint until the 2nd half of Sept at earliest, to do their assessment on feasibility and LOE. 
Workarounds available to BCBSMA users in the meantime:
1) User searches for documents generated with incorrect FMPro #  and deletes them before generating new documents
2) User searches for documents generated with incorrect FMPro # and renames them with correct FMPro # before generating new documents
3) User does not search content library for documents and only uses those downloaded directly from the SBC Preview page in the Configurator. The correct naming convention is on these documents, the issue is just when new versions of previously incorrectly-named documents are created, they take on the already-standing name of the docs. Any brand new revisions of documents are named correctly (Finals that had not yet been generated, for instance). And also, the underlying name of all new documents is correct (so if you downloaded the latest version of a document from the wrong-named docs, it will have the correct naming convention when saved locally).

**Comment by jddecoste** [comment]
Mark
I'm not seeing the value of re-hashing the original requirements as they are clearly defined and attached via illustration above. Additionally, there has been a very detailed dialog back and forth on this ticket regarding the work completed and what the limitations are in the system. The work has progressed and we are at a point where we found a limitation in the field that our users cannot make errors. We would not have listed this as an original requirement as it would not be expected that a changed field could not properly reflect that change in all areas where it populates. This expectation is pretty standard as other fields in the HR system to function off changed or corrected data.  Last week, the ask of the team was to go back and see what was possible to fix this in case errors were made inadvertently. As Michelle has clearly stated above, the breaking of the link approach is an additional enhancement and provided multiple workarounds, which unfortunately our team cannot take on as another manual step due to limitations of the systems. 

As you can imagine, the timing provided to even assess the LOE is after we've begun our Open Enrollment which is past when a fix would need to be completed to support the team. We are happy to have a meeting to brainstorm or understand if there are any other approaches that HighRoads can take fix the issue and ensure we've exhausted all possible ways to address it.
Please advise. Thanks

**Comment by markderg** [comment]
Jenn:  We would be happy to discuss this next week.  I would suggest we use our meeting Monday @ 2 pm to discuss (first half hour).  Let me know if this works for you.  Thx, Mark

**Comment by michparrish** [comment]
@g4gaurang @jddecoste @willdwyer 
The workbooks that have the new Government Website field and the new LOVs also includes this FMPro# enhancement. Has the functionality of the FMPro # field and functionality, minus the plan document naming convention issue above, been tested to satisfaction where we could move this to Production as well? Being that that will be a separate item we can tackle branched off of this field being in use, please let me know the comfort.

**Comment by g4gaurang** [comment]
@michparrish 
FMPRO# enhancement is still a  testing work in progress. Any acceptance for that will be provided separately. If you absolutely cannot decouple  these issues at the workbook level, then go ahead and move it together but it does not constitute an approval of any part of  FMPRO# enhancement issue. We need the new Government website enhancement to finish our connector plans in production.

**Comment by michparrish** [comment]
Understood. 
We will work on decoupling the configuration and testing in pshadow this evening. If we test successfully, we'll go move the workbooks without the FMPro# configuration. Adding the workbooks to prod with the FMPro# would include the new naming conventions for plan designs, impacting your entire process. We'll try to stay away from that impact as much as we can.

**Comment by markderg** [comment]
Looking for final approval BCBSMA on the original enhancement request.  Looking for an ETA on when BCBSMA will complete testing.  
Michelle P. will be sending workaround end to end process to Will D.

**Comment by g4gaurang** [comment]
This has been verified on Pclient5. Please move to PROD and notify us.

**Comment by michparrish** [comment]
All configuration has been moved to Prod. Please confirm and close.

Also, I opened item #251 to track the enhancement request branching from this one.

**Comment by willdwyer** [comment]
This has been tested in Production, and appears to be fixed.

However, note that it has been discovered that the FM job 8 character limit validation error does NOT prevent generation.  See screenshot below.

But regardless, this request is being closed.

![screen shot 2015-08-27 at 1 53 25 pm](https://cloud.githubusercontent.com/assets/13453008/9528543/0543bbaa-4cc3-11e5-85b0-816b130ab4ff.png)
